modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricer.java

```diff
@@ -0,0 +1,242 @@
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.bond;
+
+import java.time.LocalDate;
+
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.basics.currency.MultiCurrencyAmount;
+import com.opengamma.strata.basics.currency.Payment;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.market.sensitivity.PointSensitivities;
+import com.opengamma.strata.pricer.CompoundedRateType;
+import com.opengamma.strata.pricer.DiscountFactors;
+import com.opengamma.strata.pricer.DiscountingPaymentPricer;
+import com.opengamma.strata.product.bond.ResolvedBillTrade;
+
+/**
+ * Pricer for bill trades.
+ * <p>
+ * This function provides the ability to price a {@link ResolvedBillTrade}.
+ */
+public class DiscountingBillTradePricer {
+
+  /**
+   * Default implementation.
+   */
+  public static final DiscountingBillTradePricer DEFAULT = new DiscountingBillTradePricer(
+      DiscountingBillProductPricer.DEFAULT,
+      DiscountingPaymentPricer.DEFAULT);
+
+  /**
+   * Pricer for {@link ResolvedBill}.
+   */
+  private final DiscountingBillProductPricer productPricer;
+  /**
+   * Pricer for {@link Payment}.
+   */
+  private final DiscountingPaymentPricer paymentPricer;
+
+  /**
+   * Creates an instance.
+   * 
+   * @param productPricer  the pricer for {@link ResolvedBill}
+   * @param paymentPricer  the pricer for {@link Payment}
+  */
+  public DiscountingBillTradePricer(
+      DiscountingBillProductPricer productPricer,
+      DiscountingPaymentPricer paymentPricer) {
+    this.productPricer = ArgChecker.notNull(productPricer, "productPricer");
+    this.paymentPricer = ArgChecker.notNull(paymentPricer, "paymentPricer");
+  }
+
+  /**
+   * Calculates the present value of a bill trade.
+   * <p>
+   * If the settlement details are provided, the present value is the sum of the underlying product's present value
+   * multiplied by the quantity and the present value of the settlement payment if still due at the valuation date. 
+   * If not it is the underlying product's present value multiplied by the quantity.
+   * 
+   * @param bill  the bill
+   * @param provider  the discounting provider
+   * @return  the present value
+   */
+  public CurrencyAmount presentValue(ResolvedBillTrade bill, LegalEntityDiscountingProvider provider) {
+    if (provider.getValuationDate().isAfter(bill.getProduct().getNotional().getDate())) {
+      return CurrencyAmount.of(bill.getProduct().getCurrency(), 0.0d);
+    }
+    CurrencyAmount pvProduct = productPricer.presentValue(bill.getProduct(), provider)
+        .multipliedBy(bill.getQuantity());
+    if (bill.getSettlement().isPresent()) {
+      DiscountFactors discountFactorsRepo = provider.repoCurveDiscountFactors(
+          bill.getProduct().getSecurityId(), bill.getProduct().getLegalEntityId(), bill.getProduct().getCurrency())
+          .getDiscountFactors();
```

Comment:
This kind of formatting where parameters and chained method calls are at the same indent is hard to read. Move `provider.repoCurveDiscountFactors(` to a new line.